src/python/pants/backend/jvm/tasks/unpack_jars.py

```diff
@@ -62,4 +62,7 @@ def unpack_target(self, unpacked_jars, unpack_dir):
       if not unpacked_jars.payload.intransitive or coordinate in direct_coords:
         self.context.log.info('Unpacking jar {coordinate} from {jar_path} to {unpack_dir}.'.format(
           coordinate=coordinate, jar_path=jar_path, unpack_dir=unpack_dir))
-        ZIP.extract(jar_path, unpack_dir, filter_func=unpack_filter)
+        if jar_path.endswith('.jar'):
```

Comment:
So, assuming that the `unpack_filter` can be applied to `archiver_for_path` (I think it needs to be...) you shouldn't need this conditional, and can just always use `archiver_for_path`.